src/ex1/script.js

```diff
@@ -0,0 +1,165 @@
+  allTodosList.appendChild(listItem);
+  isListSorted = unsorted;
+}
+
+function createListElement(todoText, animation) {
+  const listItem = document.createElement("li");
+  if (animation) {
+    listItem.className = "add-item-animation";
+  }
+  setTimeout (() => { listItem.className = "existing-item";}, 700);
+  listItem.innerHTML = `<div class="todo-text">${todoText}</div>
+                        <button class="remove-todo-button"><i class="fa fa-trash"></i></button>`;
+  return listItem;
+}
+
+function displayFooterAndImage() {
+  displayButtonsAndAmount();
+  displayZeroImage();
+}
+
+function displayButtonsAndAmount() {
+  let tasks = "tasks";
+  if (todosArray.length === 1) {
+    tasks = "task";
+    displayElement('sort-list-button', false)
+    displayElement('clear-all-button', false)
+  } else {
+    displayElement('sort-list-button', true)
+    displayElement('clear-all-button', true)
+  }
+  amountOfTodosInfo.textContent = `${todosArray.length} pending ${tasks}`;
+}
+
+function displayZeroImage() {
+  if (todosArray.length === 0) {
+    displayElement('zero-todos-image', true)
+    displayElement('footer', false)
+  } else {
+    displayElement('zero-todos-image', false)
+    displayElement('footer', true)
+  }
+}
+
+function displayElement(elementId, toDisplay) {
+  displayStyle = toDisplay ? "" : "none";
+  document.getElementById(elementId).style.display = displayStyle;
+}
+
+function onTodoTitleClicked(clickedTodo) {
+  alert(clickedTodo.textContent);
+}
+
+function onDeleteButtonClicked(clickedButton) {
+  const index = Array.prototype.indexOf.call(allTodosList.getElementsByTagName("li"), clickedButton.parentElement);
+  todosArray.splice(index, 1);
+  clickedButton.parentElement.classList.remove("existing-item");
+  clickedButton.parentElement.classList.add("delete-item-animation");
+  setTimeout (() => {
+    clickedButton.parentElement.remove();
+    displayFooterAndImage();
+  }, 700);
+}
+
+function onClearAllButtonClicked() {
+  const deleteButtons = Array.prototype.slice.call(document.querySelectorAll("div"));
+  deleteButtons.reverse().forEach(button => {
+    onDeleteButtonClicked(button);
+  });
+}
+
+function onAddTodoFormSubmitted(event){
+  event.preventDefault();
+  const newTodoText = inputTitle.value;
+  addTodoItem(newTodoText, true);
+  todosArray.push(newTodoText);
+  displayFooterAndImage();
+  inputTitle.value = "";
+}
+
+inputTitle.addEventListener("keypress", (event) => {
+  if (event.key === "Enter") {
+    event.preventDefault();
+    document.getElementById('add-todo-button').click();
+  }
+});
+
+function onSortListButtonClicked() {
+  if (isListSorted === unsorted || isListSorted === sortedDesc) {
+    sortListWithOrder(compareElementsAsc);
+    isListSorted = sortedAsc;
+  } else {
+    sortListWithOrder(compareElementsDesc);
+    isListSorted = sortedDesc;
+  }
+}
+
+function sortListWithOrder(comparator) {
+  let switching = true;
+  let shouldSwitch = false;
```

Comment:
the sort should apply on the original todo list, and the dom just needs to reflect that order. One option is to remove all the tasks from the dom (after the sort) and re-create them in the sorted order

Reply:
It is implemented in exactly this way in the second exercise